### ISSUE TITLE
Charging Module API chore list

### ISSUE THREAD

**Issue by Cruikshanks** [DEFRA/sroc-service-team#68]
We've been hammering away getting the [sroc-charging-module-api](https://github.com/DEFRA/sroc-charging-module-api) ready for go-live.

As with all things, when facing a deadline we've parked some items that need doing in order to meet it. But we'd still like to tackle them before go live so the service is 'ship shape and Bristol fashion' before starting the next phase of its journey.

For reference, these aren't just 'niggles' that resolving would make us happy. A clean and consistent codebase is the foundation for delivering new features and functions quickly whilst maintaining quality. As more 'cruft' accrues and is not deal with, the slower our rate of delivery as a team will be.

## Test suite
- [ ] Review test helpers (merge and/or refactor where necessary)
- [ ] Determine data generator helpers required (pick tests that depend on populated bill runs for example and aim to refactor them to use the new helpers)
- [ ] Fix `test/support/helpers/transaction.helper.js` method `_newLicenceAndInvoice()` not passing on overrides (might be superceded by the other test chores
- [ ] Review controller tests to be consistent. Stub out underlying services from any that are still calling them
- [ ] Add _just one_ 'happy-path' bill run feature test
- [ ] Add _just one_ 'happy-path' customer test
- [ ] Review and resolve [SonarCloud code smells](https://sonarcloud.io/project/issues?id=DEFRA_sroc-charging-module-api&resolved=false&types=CODE_SMELL)

## Local environment
- [ ] Add ns VS Code `Nuke DB` task
- [ ] Update all tasks to use `docker compose` instead of `docker-compose`

## Code
- [ ] Services folder tidy up (grouping; bill_runs, invoices, transactions etc)
- [ ] Move to using subfolder `index.js` convention
- [ ] Squash down the DB migrations before Go live
- [ ] Convert `jwt_strategy` (authOptions) into a proper class then review #66 
- [ ] Update lib classes to include lib in name and file to match our convention

## Documentation
- [ ] Add root README to `app/` folder to explain what the purpose of each folder is, and any conventions/expectations for the files in those folders
- [ ] Update root README in `test/` with our testing conventions
- [ ] Update this project to reflect change in status of projects: charging-module-api `dead`, sroc-charging-service `redundant`, sroc-tcm-admin `maintenance-only`.
